### DISCUSSION TITLE
Attributes on primary constructors

### DISCUSSION THREAD

**Discussion by colejohnson66** [dotnet/csharplang#8043] [Language Ideas] [answered]
It is currently impossible to add attributes to a primary constructor:

```cs
[MyAttribute] // applies to entire class, not constructor
public class MyClass(int value)
{
    // ...
}
```

I propose `[ctor: XAttribute]` syntax, similar to `[return: XAttribute]` for method return values and `[assembly: XAttribute]` for assembly-level attributes.

**Comment by CyrusNajmabadi** [comment, marked as answer]
> It is currently impossible to add attributes to a primary constructor:

This is not the case.  Roslyn, as an example, places attributes on primary constructors all over the place.  You just need the `[method: ...]` syntax to specify this :)
